src/lib/EncoderHelper.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AutomatedTranslation;
+
+use DOMCdataSection;
+use DOMDocument;
+use DOMXPath;
+use Ibexa\FieldTypeRichText\FieldType\RichText\Value as RichTextValue;
+use RuntimeException;
+
+class EncoderHelper
+{
+    public function clearCDATAInTextField(string $payload): string
+    {
+        $dom = new DOMDocument();
+        $dom->loadXML($payload);
+        $xpath = new DOMXPath($dom);
+        $textNodes = $xpath->query('//text()');
+        if ($textNodes === false) {
+            return $payload;
+        }
+
+        foreach ($textNodes as $textNode) {
+            if (!$textNode instanceof DOMCdataSection) {
+                continue;
+            }
+            $parent = $textNode->parentNode;
```

Comment:
`$parent` can be a non-element node, so calling `getAttribute()` may fatally error. You can check, like above` if it is `DOMElement`